llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -1801,7 +1807,16 @@ void SpirvLowerGlobal::lowerBufferBlock() {
 
               // Run the users of the GEP to check for any nonuniform calls.
               for (User *const user : getElemPtr->users()) {
-                CallInst *const call = dyn_cast<CallInst>(user);
+                // TODO: Remove this when LLPC will switch fully to opaque pointers.
+                // TODO: revert this back to 'const'
+                CallInst *call = dyn_cast<CallInst>(user);
+
+                // TODO: Remove this when LLPC will switch fully to opaque pointers.
+                // This is needed to handle additional bitcast which was added while creating function call.
+                if (auto bitCast = dyn_cast<BitCastInst>(user)) {
+                  User *bitCastUser = bitCast->user_back();
+                  call = dyn_cast<CallInst>(bitCastUser);
+                }
```

Comment:
This makes awkward assumptions about exactly how the IR looks like at this stage. Consider what happens if `bitcast`s have been CSE'd in the meantime? There could be multiple users of the bitcast, all of which would need to be checked.

Reply:
Thanks!, now as I'm thinking about this you may be right. I will fix this !